From e2f06bbfe2b3214390567d7c3667cf6eb1f717e6 Mon Sep 17 00:00:00 2001
From: Nachi Ueno <ueno.nachi@lab.ntt.co.jp>
Date: Thu, 10 Nov 2011 10:35:57 +0900
Subject: [PATCH] Fixes bug 886263.

IndexError: pop index out of range on replace_uuid_with_id

(cherry picked from commit 10f93f567f4a164091d811765d1fdc39cdb36478)

Change-Id: I2f89bcc1675315c800e0d34bc09a268293a23caf
---
 nova/scheduler/api.py |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/nova/scheduler/api.py b/nova/scheduler/api.py
index 719437b..1705c95 100644
--- a/nova/scheduler/api.py
+++ b/nova/scheduler/api.py
@@ -342,7 +342,7 @@ class reroute_compute(object):
         """
         if 'instance_id' in kwargs:
             kwargs['instance_id'] = replacement_id
-        elif len(args) > 1:
+        elif len(args) > 2:
             args.pop(2)
             args.insert(2, replacement_id)
 
-- 
1.7.6.5

